lib/active_model/serializer.rb

```diff
@@ -102,7 +111,16 @@ def self.adapter
     end
 
     def self.root_name
-      name.demodulize.underscore.sub(/_serializer$/, '') if name
+      name.demodulize.underscore.sub(/_serialization$/, '') if name
+    end
+
+    def self.deserialize(params)
+      association_keys = _reflections.map(&:name)
+      permitted_params = adapter.params(@_params, association_keys) || @_params
+      permitted_key    = adapter.root || root_name
+
+      sanitazed_params = params.require(permitted_key.to_sym).permit(permitted_params)
```

Comment:
I would agree that the deserializer should not be responsible for determining which params are required/permitted. That is the developers responsibility. However, it would be nice to adhere to the same API that Rails provides (so in the case of a JSON API request body, parse the body and add the data to the action controller params hash).